Feature/invite url #1074

BarrySunderland opened the pull request:
PR for issue #1074 

endpoints that are used to CRUD  an inviteLinkId. This inviteLinkId is used as a proxy for the appletId, and allows users to add themselves directly to an applet. 

### New Admin-facing end points

POST    applet/ :id /inviteLink  ->  createInviteLink  
GET     applet/ :id /inviteLink  ->  getInviteLink  
PUT     applet/ :id /inviteLink  ->  replaceInviteLink  
DELETE  applet/ :id /inviteLink  ->  deleteInviteLink  

### New User-facing end points

GET     applet/invitelink/ :inviteLinkId /info  -> viewInviteLinkInfo  
    view basic info about the applet e.g.: name and description

POST    applet/invitelink/ :inviteLinkId /accept -> acceptOpenInviteLink  
    endpoint for a logged in user to add themselves to an applet
 

### Credits
This pull request is part of the "Citizen Science Logger" project and is provided by the [ETH Library Lab](https://www.librarylab.ethz.ch/)

jj105 commented:
@BarrySunderland thanks for PR.
I reviewed changes. it looks good to me.

I have few things to discuss.
1 - can we add appropriate index for invitation_id to optimize search?
( I think you might need to update initialize function in models/folder.py )
2 - can we remove commented lines in acceptOpenInviteLink function if they won't be used/considered in the future?

BarrySunderland commented:
> @BarrySunderland thanks for PR.
> I reviewed changes. it looks good to me.
> 
> I have few things to discuss.
> 1 - can we add appropriate index for invitation_id to optimize search?
> ( I think you might need to update initialize function in models/folder.py )
> 2 - can we remove commented lines in acceptOpenInviteLink function if they won't be used/considered in the future?

Thanks for reviewing this. 
1. I added invitation.id like you said and just confirmed that it is now included by checking `db.folder.getIndexes()`
2. Apologies for this, I had already removed these lines but somehow didn't include it in the commits for the PR.

One last point is that I've used the first 18 characters of a uuid4 to create the invitelink.id  
There could be a case for using `bson.objectid.ObjectId` instead, but it would make the public facing url much longer. So I have left it as uuid for now.

devbtech commented:
@BarrySunderland Could you please fix conflicts here?
